webapp/channels/src/components/move_thread_modal/move_thread_modal.tsx

```diff
@@ -0,0 +1,256 @@
+
+import {makeGetChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
+import NotificationBox from 'components/notification_box';
+
+import {GlobalState} from 'types/store';
+
+import Constants from 'utils/constants';
+
+import PostMessagePreview from 'components/post_view/post_message_preview';
+import {GenericModal} from '@mattermost/components';
+
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import {Post, PostPreviewMetadata} from '@mattermost/types/posts';
+import {Channel} from '@mattermost/types/channels';
+
+import ForwardPostChannelSelect, {ChannelOption} from '../forward_post_modal/forward_post_channel_select';
+
+import './move_thread_modal.scss';
+import {ClientError} from '@mattermost/client';
+
+export type ActionProps = {
+
+    // join the selected channel when necessary
+    joinChannelById: (channelId: string) => Promise<ActionResult>;
+
+    // switch to the selected channel
+    switchToChannel: (channel: Channel) => Promise<ActionResult>;
+
+    // action called to move the post from the original channel to a new channel
+    moveThread: (postId: string, channelId: string) => Promise<ActionResult>;
+}
+
+export type OwnProps = {
+
+    // The function called immediately after the modal is hidden
+    onExited?: () => void;
+
+    // the post that is going to be moved
+    post: Post;
+};
+
+export type Props = OwnProps & {actions: ActionProps};
+
+const noop = () => {};
+
+const getChannel = makeGetChannel();
+
+const MoveThreadModal = ({onExited, post, actions}: Props) => {
+    const {formatMessage} = useIntl();
+
+    const originalChannel = useSelector((state: GlobalState) => getChannel(state, {id: post.channel_id}));
+    const currentTeam = useSelector(getCurrentTeam);
+
+    const [bodyHeight, setBodyHeight] = useState<number>(0);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [postError, setPostError] = useState<React.ReactNode>(null);
+    const [selectedChannel, setSelectedChannel] = useState<ChannelOption>();
+    const [isButtonClicked, setIsButtonClicked] = useState<boolean>(false);
+
+    const bodyRef = useRef<HTMLDivElement>();
+
+    const measuredRef = useCallback((node) => {
+        if (node !== null) {
+            bodyRef.current = node;
+            setBodyHeight(node.getBoundingClientRect().height);
+        }
+    }, []);
+
+    const onHide = useCallback(() => {
+        onExited?.();
+    }, [onExited]);
+
+    const handleChannelSelect = useCallback(
+        (channel: ValueType<ChannelOption>) => {
+            if (Array.isArray(channel)) {
+                setSelectedChannel(channel[0]);
+            }
+            setSelectedChannel(channel as ChannelOption);
+        },
+        [],
+    );
+
+    // since the original post has a click handler specified we should prevent any action here
+    const preventActionOnPreview = (e: React.MouseEvent) => {
```

Comment:
You are passing this to a component prop, so you are passing a new function on every render. Since you don't need anything form the component, you can define this function outside of it, so it is only declared once.